docs/security.md

```diff
@@ -1,60 +1,259 @@
+    | `issuerUrl` | URL of the authentication provider which allows the Pulsar client to obtain an access token | Required | `https://accounts.google.com` |
+    | `audience`  | An OAuth 2.0 "resource server" identifier for the Pulsar cluster | Required |`https://broker.example.com` | 
 
-props.put("security.protocol", "SASL_PLAINTEXT");
-props.put("sasl.mechanism", "PLAIN");
-```
 
-Kafka consumers and Kafka producers can use the props to connect to brokers.
+    ```properties
+    authenticationEnabled=true
+    authorizationEnabled=true
+    authenticationProviders=org.apache.pulsar.broker.authentication.AuthenticationProviderToken
+    superUserRoles=<super-user-roles>
+    brokerClientAuthenticationPlugin=org.apache.pulsar.client.impl.auth.oauth2.AuthenticationOAuth2
+    brokerClientAuthenticationParameters={"type":"client_credentials","privateKey":"file:///path/to/credentials_file.json","issuerUrl":"<issuer-url>","audience":"<audience>"}
+    tokenPublicKey=<token-public-key>
+    ``` 
+
+2. Enable authentication on KoP.
+
+    (1) Set the following property in the `conf/broker.conf` or `conf/standalone.conf` file.
+
+    ```properties
+    saslAllowedMechanisms=OAUTHBEARER
+    ```
+
+    (2) Specify the Kafka server callback handler.
+
+    For the `OAUTHBEARER` mechanism, you can use `AuthenticationProviderToken` or custom your authentication provider to process the access tokens from OAuth 2.0 server.
+
+    KoP provides a built-in `AuthenticateCallbackHandler` that uses the authentication provider of Pulsar for authentication. You need to configure the following properties in the `conf/kop-handler.properties` file.
+
+    ```properties
+    # Use the KoP's built-in handler 
+    kopOauth2AuthenticateCallbackHandler=io.streamnative.pulsar.handlers.kop.security.oauth.OauthValidatorCallbackHandler
+
+    # Java property configuration file of OauthValidatorCallbackHandler
+    kopOauth2ConfigFile=conf/kop-handler.properties
+    ```
+
+(3) Specify the authentication method name of the provider (that is, `oauth.validate.method`) in the `conf/kop-handler.properties` file.
+
+   - If you use `AuthenticationProviderToken`, since `AuthenticationProviderToken#getAuthMethodName()` returns `token`, set the `oauth.validate.method` as the token.
+
+   - If you use other providers, set the `oauth.validate.method` as the result of `getAuthMethodName()`.
+
+        ```properties
+        oauth.validate.method=token
+        ```
+
+3. Enable authentication on Kafka client.
+
+    (1) Install the KoP built-in callback handler to your local Maven repository.
+
+    To get an access token from an OAuth 2.0 server, you need to use the KoP built-in callback handler instead of the Kafka login callback handler.
+
+    ```bash
+    mvn clean install -pl oauth-client -DskipTests
+    ```
+
+    (2) Add the following dependencies to the `pom.xml` file.
+
+    For stable releases, the `pulsar.version` is the same to the `kop.version`.
```

Comment:
the same as?

Reply:
![image](https://user-images.githubusercontent.com/50226895/112241387-1c674100-8c85-11eb-8c61-74e0e2dcdf5f.png)

Think `same to` is OK